app/scripts/controllers/build.js

```diff
@@ -67,72 +126,17 @@ angular.module('openshiftConsole')
-            setLogVars(build);
-            var buildNumber = $filter("annotation")(build, "buildNumber");
-            if (buildNumber) {
-              $scope.breadcrumbs[2].title = "#" + buildNumber;
-            }
 
+        DataService
+          .get("builds", $routeParams.build, context)
+          .then(function(build) {
+            buildResolved(build);
             // If we found the item successfully, watch for changes on it
-            watches.push(DataService.watchObject("builds", $routeParams.build, context, function(build, action) {
-              if (action === "DELETED") {
-                $scope.alerts["deleted"] = {
-                  type: "warning",
-                  message: "This build has been deleted."
-                };
-              }
-              $scope.build = build;
-              setLogVars(build);
-            }));
-            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, function(buildConfig, action) {
-              if (action === "DELETED") {
-                $scope.alerts["deleted"] = {
-                  type: "warning",
-                  message: "Build configuration " + $scope.buildConfigName + " has been deleted."
-                };
-              }
-              $scope.buildConfig = buildConfig;
-              $scope.paused = BuildsService.isPaused($scope.buildConfig);
-              updateCanBuild();
-            }));
-          },
-          // failure
-          function(e) {
-            $scope.loaded = true;
-            $scope.alerts["load"] = {
-              type: "error",
-              message: "The build details could not be loaded.",
-              details: "Reason: " + $filter('getErrorDetails')(e)
-            };
-          }
-        );
-
-        watches.push(DataService.watch("builds", context, function(builds, action, build) {
-          $scope.builds = {};
-          // TODO we should send the ?labelSelector=buildconfig=<name> on the API request
-          // to only load the buildconfig's builds, but this requires some DataService changes
-          var allBuilds = builds.by("metadata.name");
-          angular.forEach(allBuilds, function(build, name) {
-            if (build.metadata.labels && build.metadata.labels.buildconfig === $routeParams.buildconfig) {
-              $scope.builds[name] = build;
-            }
-          });
-
-          var buildConfigName;
-          var buildName;
-          if (build) {
-            buildConfigName = build.metadata.labels.buildconfig;
-            buildName = build.metadata.name;
-          }
+            watches.push(DataService.watchObject("builds", $routeParams.build, context, buildResolved));
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, buildConfigResolved));
+          }, buildRejected);
 
-          updateCanBuild();
-        }));
+        watches.push(DataService.watch("builds", context, buildsResolved));
```

Comment:
Oh nice.  Yeah, I didn't even look at that, was just going for a simple ctrl to prototype sharing the get/watch callback but cutting more code is a win! :smile: